pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -191,26 +191,50 @@ func BuildListenerTLSContext(serverTLSSettings *networking.ServerTLSSettings,
 		}
 	}
 
-	// Set TLS parameters if they are non-default
-	var ecdhCurves []string
-	if mesh != nil && mesh.TlsDefaults != nil &&
-		(serverTLSSettings.Mode == networking.ServerTLSSettings_MUTUAL || serverTLSSettings.Mode == networking.ServerTLSSettings_SIMPLE) {
-		ecdhCurves = mesh.TlsDefaults.EcdhCurves
-	}
-	if len(serverTLSSettings.CipherSuites) > 0 || len(ecdhCurves) > 0 ||
-		serverTLSSettings.MinProtocolVersion != networking.ServerTLSSettings_TLS_AUTO ||
-		serverTLSSettings.MaxProtocolVersion != networking.ServerTLSSettings_TLS_AUTO {
-		ctx.CommonTlsContext.TlsParams = &auth.TlsParameters{
-			TlsMinimumProtocolVersion: convertTLSProtocol(serverTLSSettings.MinProtocolVersion),
-			TlsMaximumProtocolVersion: convertTLSProtocol(serverTLSSettings.MaxProtocolVersion),
-			CipherSuites:              serverTLSSettings.CipherSuites,
-			EcdhCurves:                ecdhCurves,
-		}
+	if isSimpleOrMutual(serverTLSSettings.Mode) {
+		// If Mesh TLSDefaults are set, use them.
+		applyDownstreamTLSDefaults(mesh.GetTlsDefaults(), ctx.CommonTlsContext)
+		applyServerTLSSettings(serverTLSSettings, ctx.CommonTlsContext)
 	}
-
 	return ctx
 }
 
+func applyDownstreamTLSDefaults(tlsDefaults *meshconfig.MeshConfig_TLSConfig, ctx *auth.CommonTlsContext) {
+	if tlsDefaults == nil {
+		return
+	}
+
+	if len(tlsDefaults.EcdhCurves) > 0 {
+		tlsParamsOrNew(ctx).EcdhCurves = tlsDefaults.EcdhCurves
+	}
+	if tlsDefaults.MinProtocolVersion != meshconfig.MeshConfig_TLSConfig_TLS_AUTO {
```

Comment:
IMO, we donot need to chweck TLS_AUTO\

And why not set tlsMinVersion in upstream cluster `applyTLSDefaults`

Reply:
> IMO, we donot need to chweck TLS_AUTO\

This indicates user has actually set the value in the enum since TLS_AUTO is default otherwise we should use default TLS version and not set AUTO

> And why not set tlsMinVersion in upstream cluster applyTLSDefaults

I think that is a separate discussion we need to have. Currently TLS version setting is only enforced on server (services having sidecar). If we set it on upstream cluster - we are forcing version on external services and it will break lot of people.

Reply:
Not a big concern, just from api view we do not distinguish inbound and outbound